File: static/code/sample/tethered/gdl-hijoystkl-c.cs
// In this sample:
// Update X, Y and Button status every 100ms

using System;
using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

var button_pin = (int)duelink.Engine.ExecuteCommand("BtnPin()");

int ReadX() {
    var ret = duelink.Engine.ExecuteCommand("JoystickX()");
    return (int)ret;
}
int ReadY() {
    var ret = duelink.Engine.ExecuteCommand("JoystickY()");
    return (int)ret;
}

int ReadButton() {
    var ret = duelink.Digital.Read(button_pin, DUELinkController.InputType.PullUp);
    return ret == false ? 0:1;
}

while (true) {
    var x = ReadX();
    var y = ReadY();
    var bt = ReadButton();

    Console.WriteLine($"X = {x}, Y = {y}, Button status: {bt}");

    Thread.Sleep(100);
}

File: static/code/sample/tethered/gdl-hijoystkl-c.py
# In this sample:
# Update X, Y and Button status every 100ms

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
button_pin = int(duelink.Engine.ExecuteCommand("BtnPin()"))

def ReadX():
    ret = duelink.Engine.ExecuteCommand("JoystickX()")
    return int(ret)

def ReadY():
    ret = duelink.Engine.ExecuteCommand("JoystickY()")
    return int(ret)

def ReadButton():
    ret = duelink.Digital.Read(button_pin, 1)
    return 0 if ret == False else 1

while True:
    x = ReadX()
    y = ReadY()
    bt = ReadButton()

    print(f"X = {x}, Y = {y}, Button status: {bt}")

    time.sleep(0.1)